tests/Microsoft.Identity.Test.Integration.net45/SeleniumTests/ConfidentialClientAuthorizationTests.cs

```diff
@@ -75,6 +75,17 @@ public async Task SeleniumGetAuthCode_RedeemForAt_CommonAuthority_Async()
             await RunTestForUserAsync(labResponse, $"https://login.microsoftonline.com/{TenantId}").ConfigureAwait(false);
         }
 
+        [TestMethod]
+        // Regression test for: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/891
+        public async Task SeleniumGetAuthCode_RedeemForAt_CommonAuthority_MsaUser_Async()
```

Comment:
Is it really worth running this rather expensive test with an MSA user? (genuine question) I think we want to make sure happy paths for all our flows work with MSA users, but need to exercise some causion around old regression tests.

Reply:
we can discuss.

---
In reply to: [344433906](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1496#discussion_r344433906) [](ancestors = 344433906)

Reply:
I would like to see the numbers telling us more precisely how expensive it really is before optimizing.

---
In reply to: [344960019](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1496#discussion_r344960019) [](ancestors = 344960019,344433906)

Reply:
An interactive login takes with Selenium takes about 5 seconds. Most of our tests are more complex, they perform the flow serveral times etc. Test time does not vary too much: 

![image](https://user-images.githubusercontent.com/12273384/68751308-458bb200-05f9-11ea-8dc7-53af37d40e2b.png)

Entire Integration test suite: ~3 min on .net 45, ~2 min on .net core
We don't seem to be running the .net47 and .net48 tests :( I'll fix this.

Reply:
The particular test you are pointing at is a pretty complex one, it takes 30 seconds (our most expensive test)